db/migrate/20150831193037_move_observations_to_roles.rb

```diff
@@ -5,11 +5,21 @@ def change
 
     # We have to use raw sql because we've subclassed the observation class
     pgres = ProposalRole.connection.execute("SELECT proposal_id, user_id from observations")
+    seen_proles = {}
```

Comment:
Minor: should `seen_proles` be a set?

A bigger concern is the n+1 queries. Even though this is only run once, each migration has a timeout (due to being included in the cloud foundry start command).
What do you think about
```SQL
SELECT proposal_id, user_id, users.id FROM observations LEFT JOIN users ON (user_id = users.id)
```
Then if the third column were null, we'd know the user doesn't exist?

You could also group by proposal_id and user_id (or `DISTINCT`) if you wanted to get rid of duplicates via SQL rather than in Ruby land.